lib/puppet/provider/service/systemd.rb

```diff
@@ -46,11 +67,22 @@ def status
   end
 
   def enable
+    output = systemctl("unmask", @resource[:name])
```

Comment:
I don't know if one way or the other is more correct; throwing an error in this case might not provide a way to unmask a masked service, so this might be the only viable approach as-is. I think this approach will do.